src/pages/publication/DeletePublicationModal.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { FC, useState } from 'react';
+import Modal from './../../components/Modal';
+import { useTranslation } from 'react-i18next';
+import { DialogActions, DialogContent, Button } from '@material-ui/core';
+import styled from 'styled-components';
+import NormalText from './../../components/NormalText';
+
+const StyledTitleContainer = styled.div`
+  font-style: italic;
+`;
+
+interface DeletePublicationModalProps {
+  id: string;
+  title: string;
+  setDeletePublicationId: (id: string) => void;
```

Comment:
fjern denne propen, og overlat vis/skjul modal helt til `openModal` i state?

Reply:
Ser ikke helt hvordan jeg skal få til dette siden jeg bruker en komponent for alle innslag i tabellen

Reply:
virker som at denne bare trengs for å sette blank id, som brukes for å vite at modalen ikke skal vises. kunne vi ikke bare brukt `setOpenModal(false)` internt i denne komponent for det samme?

Reply:
```suggestion
  const [openModal, setOpenModal] = useState(false);

  const handleOnClick = (publication: PublicationPreview) => {
    setOpenModal(true);
    setDeletePublicationId(publication.id);
    setDeletePublicationTitle(publication.title);
  };

  return (
    <>
      <StyledTable>
 ...
      </StyledTable>
      {openModal && (
        <DeletePublicationModal id={deletePublicationId} title={deletePublicationTitle} setOpenModal={setOpenModal} />
      )}
    </>
  );
```

og deretter;

```suggestion
interface DeletePublicationModalProps {
  id: string;
  title: string;
  setOpenModal: (open: boolean) => void;
}

const DeletePublicationModal: FC<DeletePublicationModalProps> = ({ id, title, setOpenModal }) => {
  const { t } = useTranslation();

  const cancelDelete = () => {
    setOpenModal(false);
  };

  const deletePublication = () => {
    // delete publication here
    setOpenModal(false);
  };

  return (
    <Modal openModal headingText={t('workLists:delete_publication_heading')}>
```

Reply:
Du kan flytte tilstanden et hakk opp til PublicationList, også holder den styr på om modalen skal åpnes eller ikke

Reply:
Klarte nesten å komme dit på egenhånd :)